lib/src/rules/use_build_context_synchronously.dart

```diff
@@ -17,11 +17,13 @@ const _details = r'''
 **DON'T** use BuildContext across asynchronous gaps.
 
 Storing `BuildContext` for later usage can easily lead to difficult to diagnose
-crashes. Asynchronous gaps are implicitly storing `BuildContext` and are some of
-the easiest to overlook when writing code.
+crashes, because it can change its state during the async gap and no longer be 
```

Comment:
instead of talking about changing state, let's just say that the BuildContext becomes invalid because it may no longer be associated with an Element. A `BuildContext` doesn't have a state per-se.